src/prog_models/sim_result.py

```diff
@@ -181,21 +181,27 @@ def remove(self, d = None, t = None, s = None) -> None:
             t: Time value to be removed.
             s: State value to be removed.
         """ 
-        if (not [i for i in (d,t,s) if i is not None or None]):
+        if (not [i for i in (d,t,s) if i is not None or None]) or (not [i for i in (d,t) if i is None]) or (not [i for i in (t,s) if i is None]) or (not [i for i in (d,s) if i is None]):
```

Comment:
Perhaps there is a better way to check this?

Its checking to make sure that (d,t,s) representing the (data,times,state) targets meet these conditions:
- All three cannot be specified
- All three cannot be None
- Two out of the three are specified (this specifically - could we improve somehow? Currently using 3 list comprehensions to check all three combinations of (d,t), (d,s), (t,s))

Reply:
hmmm... probably is a better way, but this can be tricky. How about:

```python
if sum([i is None for i in (d, t, s)]) != 1:
```

That would count how many variables are not none (should only be one). would that work?